Userland/Libraries/LibSQL/Heap.h

```diff
@@ -23,46 +61,41 @@ namespace SQL {
  *
  * A Heap can be thought of the backing storage of a single database. It's
  * assumed that a single SQL database is backed by a single Heap.
- *
- * Currently only B-Trees and tuple stores are implemented.
  */
 class Heap : public Core::Object {
     C_OBJECT(Heap);
 
 public:
-    static constexpr u32 VERSION = 3;
-    static constexpr u32 BLOCK_SIZE = 1024;
+    static constexpr u32 VERSION = 4;
 
     virtual ~Heap() override;
 
     ErrorOr<void> open();
-    u32 size() const { return m_end_of_file; }
-    ErrorOr<ByteBuffer> read_block(u32);
-    [[nodiscard]] u32 new_record_pointer();
-    [[nodiscard]] bool valid() const { return static_cast<bool>(m_file); }
+    bool has_block(Block::Index) const;
+    [[nodiscard]] Block::Index request_new_block_index() { return m_next_block++; }
 
     u32 schemas_root() const { return m_schemas_root; }
 
     void set_schemas_root(u32 root)
     {
         m_schemas_root = root;
-        update_zero_block();
+        MUST(update_zero_block());
```

Comment:
The `MUST`s added here and below feel like they should be `.release_value_but_fixme....`

Reply:
They are subsequently removed in the commit after, but I agree - should have been the `.release_value...` kind!

Reply:
> They are subsequently removed in the commit after

Are they? This is the 2nd to last commit, the next (last) commit is the pointer -> block_index rename, doesn't seem to remove this MUST

Reply:
Oh no they weren't - originally I had planned for this PR to also contain a huge fallibility update, but that basically exploded into way uglier code than I was willing to accept, so I left that commit out and forgot to update this one.

Reply:
Fixed!

Reply:
ah yeah I tried it too when I wanted to port SQL::Value to String, lost motivation pretty quick trying to get errors propagated. Would be nice if we could sort of "unwind" this code a bit.